__tests__/integration/send-tx.test.js

```diff
@@ -0,0 +1,1005 @@
+      })
+      .set({ 'x-wallet-id': wallet2.walletId });
+
+    expect(response.status).toBe(200);
+    expect(response.body.hash).toBeUndefined();
+    expect(response.body.success).toBe(false);
+    done();
+  });
+
+  it('should reject for insufficient funds with two outputs', async done => {
+    // Both outputs are below the 1000 HTR available
+    const response = await TestUtils.request
+      .post('/wallet/send-tx')
+      .send({
+        outputs: [
+          {
+            address: await wallet2.getAddressAt(1),
+            value: 800
+          },
+          {
+            address: await wallet2.getAddressAt(2),
+            value: 800
+          },
+        ],
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    expect(response.status).toBe(200);
+    expect(response.body.hash).toBeUndefined();
+    expect(response.body.success).toBe(false);
+    done();
+  });
+
+  it('should reject for insufficient funds with two inputs', async done => {
+    // Both inputs are have only 2000 HTR
+    const response = await TestUtils.request
+      .post('/wallet/send-tx')
+      .send({
+        inputs: [fundTx2, fundTx3],
+        outputs: [
+          {
+            address: await wallet2.getAddressAt(1),
+            value: 3000
+          },
+        ],
+      })
+      .set({ 'x-wallet-id': wallet3.walletId });
+
+    expect(response.status).toBe(200);
+    expect(response.body.hash).toBeUndefined();
+    expect(response.body.success).toBe(false);
+    done();
+  });
+
+  it('should reject for insuficcient funds on queryAddress', async done => {
+    // Wallet1 has enough funds, but none of them are on index 5
+    const response = await TestUtils.request
+      .post('/wallet/send-tx')
+      .send({
+        inputs: [{
+          type: 'query',
+          filter_address: await wallet1.getAddressAt(5)
+        }],
+        outputs: [{
+          address: await wallet2.getAddressAt(0),
+          value: 10
+        }],
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    expect(response.status).toBe(200);
+    expect(response.body.hash).toBeUndefined();
+    expect(response.body.success).toBe(false);
+    done();
+  });
+
+  it('should reject for insuficcient funds on input', async done => {
+    const response = await TestUtils.request
+      .post('/wallet/send-tx')
+      .send({
+        inputs: [fundTx1],
+        outputs: [{
+          address: await wallet2.getAddressAt(0),
+          value: 1001
+        }],
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    expect(response.status).toBe(200);
+    expect(response.body.hash).toBeUndefined();
+    expect(response.body.success).toBe(false);
+    done();
+  });
+
+  // Lastly, testing success cases, which have side-effects
+
+  it('should send with only the output address and value', async done => {
+    const tx = await wallet1.sendTx({
+      fullObject: {
+        outputs: [{
+          address: await wallet2.getAddressAt(0),
```

Comment:
Just like I said in the other files, I feel we should always check that the amount was sent to the correct address. When using address and when using change address. This applies to all tests below as well

Reply:
Done!